scripts/testing/programs/match_template/test_template_reproducibility.py

```diff
@@ -0,0 +1,265 @@
+from os import makedirs
+import cistem_test_utils.args as tmArgs
+import cistem_test_utils.make_tmp_runfile as mktmp
+import cistem_test_utils.run_job as runner
+import cistem_test_utils.temp_dir_manager as temp_dir_manager
+import mrcfile
+import numpy as np
+import tempfile
+import os
+import shutil
+import sys
+import re
+
+# By default the "_gpu" suffix will be added unless the --old-cistem flag is used
+# or the --cpu flag is used
+wanted_binary_name = 'match_template'
+
+
+def extract_threshold_value(hist_file_path):
+    """
+    Extract the threshold value from the histogram text file.
+
+    The threshold value is in the first line of the file, which starts with
+    "# Expected threshold = ". This function extracts the numerical value following this marker.
+
+    Args:
+        hist_file_path (str): Path to the histogram text file
+
+    Returns:
+        float: The extracted threshold value
+
+    Raises:
+        FileNotFoundError: If the file doesn't exist
+        ValueError: If the threshold value cannot be found or parsed
+    """
+    if not exists(hist_file_path):
+        raise FileNotFoundError(f"Histogram file not found at {hist_file_path}")
+
+    with open(hist_file_path, 'r') as f:
+        first_line = f.readline().strip()
+
+    # Use regex to extract the threshold value from the line
+    # Looking for a pattern like "# Expected threshold = 6.90"
+    match = re.search(r'# Expected threshold = ([\d.e+-]+)', first_line)
+    if not match:
+        raise ValueError(f"Could not find threshold value in {hist_file_path}")
+
+    threshold_value = float(match.group(1))
+    return threshold_value
+
+
+def main():
+    try:
+        # Parse command-line arguments
+        args = tmArgs.parse_TM_args(wanted_binary_name)
+
+        # Handle temp directory management options
+        if args.list_temp_dirs:
+            temp_dir_manager.print_temp_dirs()
+            return 0
+
+        if args.rm_temp_dir is not None:
+            success, message = temp_dir_manager.remove_temp_dir(args.rm_temp_dir)
+            print(message)
+            return 0 if success else 1
+
+        if args.rm_all_temp_dirs:
+            success_count, failure_count = temp_dir_manager.remove_all_temp_dirs()
+            print(f"Successfully removed {success_count} temporary directories.")
+            if failure_count > 0:
+                print(f"Failed to remove {failure_count} temporary directories.")
+            return 0 if failure_count == 0 else 1
+
+        # Create a temporary directory to store our replicate MIPs and track it
+        temp_dir = temp_dir_manager.create_temp_dir(prefix="template_match_reproducibility_")
+        print(f"Temporary directory created at: {temp_dir}")
+
+        # We'll run template matching 3 times to generate replicates
+        elapsed_time = [0, 0, 0]
+        mip_filenames = []
+        hist_filenames = []
+        threshold_values = []
+
+        # Run the template matching 3 times
```

Comment:
Update this block now that we have 4 replicates. Even better, let's set a variable at the top of this method that defines how many replicates we want, that way if we change the number in the future it is not hardcoded.